ext/spl/spl_fixedarray.c

```diff
@@ -356,23 +356,32 @@ static zend_long spl_offset_convert_to_long(zval *offset) /* {{{ */
 	return 0;
 }
 
-static zval *spl_fixedarray_object_read_dimension_helper(spl_fixedarray_object *intern, zval *offset)
+static zend_always_inline zend_ulong spl_offset_convert_to_ulong(zval *offset)
 {
-	zend_long index;
+	if (EXPECTED(Z_TYPE_P(offset) == IS_LONG)) {
+		/* Allow skipping exception check at call-site. */
+		ZEND_ASSERT(!EG(exception));
+		return Z_LVAL_P(offset);
```

Comment:
Isn't this a problem if the index is less than 0 tho? Or is the assumption that a negative number will be converted to such a large positive integer which is certain to fail the bound check?

If so please add a comment just so people after us can understand this.

Reply:
Yes indeed, I added a comment saying this now:

> Returned index is an unsigned number such that we don't have to do a negative check.
>  Negative numbers will be mapped at indices larger than ZEND_ULONG_MAX,
>  which is beyond the maximum length.